### PR TITLE
performance fix: do not call RNG for testing failures when not needed

### PR DESCRIPTION
**Pull request by ma2bd**
## Motivation

Small performance improvement

## Proposal

Fix the condition to skip the RNG if the failure is 0 (as expected in production).

## Test Plan

CI

### CODE REVIEW

File: linera-rpc/src/grpc_network.rs [hunk 1]
```diff
@@ -356,7 +356,9 @@ where
         let pool = ConnectionPool::default();
 
         while let Some((cross_chain_request, shard_id)) = receiver.next().await {
-            if rand::thread_rng().gen::<f32>() < cross_chain_sender_failure_rate {
+            if cross_chain_sender_failure_rate > 0.0
+                && rand::thread_rng().gen::<f32>() < cross_chain_sender_failure_rate
```
Comment (afck) [inline linera-rpc/src/grpc_network.rs:360]: We could use [`gen_bool`](https://docs.rs/rand/latest/rand/trait.Rng.html#method.gen_bool) here (same in `simple_network`):

```suggestion
                && rand::thread_rng().gen_bool(cross_chain_sender_failure_rate)
```

**Review by afck — APPROVED** [review]

File: linera-rpc/src/grpc_network.rs (hunk 1 above)
Comment (ma2bd) [reply linera-rpc/src/grpc_network.rs:360]: Nice!

Comment (ma2bd) [reply linera-rpc/src/grpc_network.rs:360]: From what what I gather, `gen_bool(1.0)` is optimized but not `gen_bool(0.0)`.

Comment (ma2bd) [reply linera-rpc/src/grpc_network.rs:360]: Incidentally, `gen_bool` uses a `f64` so I'm going to stick with `gen::<f32>`.

**Pull request merged**